runner/src/server/plugins/engine/views/components/freetextfield.html

```diff
@@ -0,0 +1,96 @@
+{% macro FreeTextField(component) %}
+
+<head>
+  <script src="/assets/tinymce.min.js"></script>
+  <script type="text/javascript">
+
+    tinymce.init({
+      setup: function (editor) {
+        editor.on('init', function () {          
+          setWordCount(editor);
+        });
+        editor.on('input', function (e) {          
+          setWordCount(editor);
+        });
+        editor.on('setcontent', function(e) {
+          // When pasting into the field, update the word count         
+          setWordCount(editor);
+        });
+      },
+      selector: '.myeditablediv', 
+      skin: 'tinymce-5',
+      branding: false,
+      elementpath: false,
+      content_css: 'default',
+      height: 150,
+      menubar: false,      
+      browser_spellcheck: true,
+      max_words: Number('{{ component.model.maxwords }}'),
+      iframe_aria_text: '{{ component.model.name }}',
+      paste_data_images: false,
+      dragover_callback: function (e) {
+        // prevent dragover event from doing anything, images could be snuck in this way
+        e.preventDefault();
+      },
+      drop_callback: function (e) {
+        // prevent drop event from doing anything, images could be snuck in this way
+        e.preventDefault();
+      },
+      plugins: [
+        'advlist', 'autolink',
+        'lists', 'link', 'charmap', 'preview', 'anchor', 'searchreplace',
+        'fullscreen', 'insertdatetime', 'table', 'help'
+      ],
+      toolbar: 'bold | bullist numlist',
+      contextmenu: 'paste'
+    });
+
+    function countWords(editor) {      
+      var content = editor.getContent({ format: 'text' });
+      // removing line breaks ast they were sometimes counting as words
+      var noLineBreaksContent = content.replace(/(\r\n|\n|\r)/gm, " ");
+      var wordCount = 0;
+
+      if (content) {
+        // Regex for checking white space border and not counting special characters
+        var matches = noLineBreaksContent.match(/\b\w+\b/g);
+        wordCount = matches ? matches.length : 0;
+      }
+      return wordCount;
+    }
+
+    function setWordCount(editor) {            
+      var maxWords = editor.getParam('max_words', 500);
+      var wordCount = countWords(editor);
+      var wordsRemaing = Number(maxWords) - wordCount;
+      var label = document.getElementById('word-count-label-' + editor.id);
+      label.innerHTML = 'You have ' + wordsRemaing + ' words remaining';
+      label.style.color = '#505a5f';
+
+      if (wordsRemaing < 0) {
+        var wordsOver = wordCount - Number(maxWords);
+        label.innerHTML = 'You have ' + wordsOver + ' words too many';
```

Comment:
This is only a string, so not a big deal, but generally I think `textContent` is preferred for plain text